### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by xpol**
Add package-lock.json as yarn is not included in the latest node versions

### CODE REVIEW

**Comment by xpol** [comment]
@xavierchow @makara @bsdelf @MiffyLiye Can we test, merge this PR and release a new npm package?

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import typescriptEslintEslintPlugin from "@typescript-eslint/eslint-plugin";
```
Comment (xavierchow) [inline eslint.config.mjs:1]: what's the difference between eslintrc and eslitn.config.mjs?

File: package.json [hunk 2]
```diff
@@ -30,35 +30,40 @@
...
   "dependencies": {
-    "@nestjs/common": "^9.3.12",
-    "@nestjs/core": "^9.3.12",
-    "ottoman": "^2.3.2",
-    "reflect-metadata": "^0.1.13",
-    "rxjs": "^7.8.0"
+    "reflect-metadata": "^0.2.2",
+    "rxjs": "^7.8.1"
+  },
+  "peerDependencies": {
```
Comment (xavierchow) [inline package.json:36]: 👍 
can you help to add `engines.code` to make sure node version >=18.

I recall npm behaviour about peerDependencies changed around npm v7(probably node v16~ish).

File: package.json [hunk 3]
```diff
@@ -76,5 +81,6 @@
     ],
     "coverageDirectory": "../coverage",
     "testEnvironment": "node"
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
```
Comment (xavierchow) [inline package.json:85]: is it necessary to specify the hash here?

**Comment by xavierchow** [comment]
@xpol I left some comments about the code changes, I don't see major issues for this PR.

File: eslint.config.mjs (hunk 1 above)
Comment (xpol) [reply eslint.config.mjs:1]: It auto converted by the `npx @eslint/migrate-config .eslintrc` command. see https://eslint.org/docs/latest/use/configure/migration-guide.

File: package.json (hunk 2 above)
Comment (xpol) [reply package.json:39]: ✅️

File: package.json (hunk 3 above)
Comment (xpol) [reply package.json:88]: Those lines are generated by `npm install`.
I'd rather remove yarn if you prefer.
Yan is an abandoned project. It's not supported anymore and not included in node distribution now.

Comment (xpol) [reply package.json]: removed

**Review by xavierchow — APPROVED** [review]

**Comment by xpol** [comment]
@xavierchow Can we publish a new npm version for this update?

**Comment by xavierchow** [comment]
@xpol published v3.1.0.

**Pull request merged**